File: practice-challenges/find-longest/find_longest.py
"""Solution for Find Longest Word."""


def get_longest_word(words):
    longest_word = ""

    for word in words:
        if len(word) > len(longest_word):
            longest_word = word

    return longest_word


# Tests

# ["boo", "a", "I", "hi"] => 'boo'
print(get_longest_word(["boo", "a", "I", "hi"]))

# ["one", "two", "superduper", "dog"] => 'superduper'
print(get_longest_word(["one", "two", "superduper", "dog"]))

# ["hello", "world"] => 'hello'
print(get_longest_word(["hello", "world"]))

File: practice-challenges/find-longest/findLongest.js
/* Solution for Find Longest Word. */

function getLongestWord(words) {
  let longestWord = "";

  for (let word of words) {
    if (word.length > longestWord.length) {
      longestWord = word;
    }
  }

  return longestWord;
}

// Tests

// ["boo", "a", "I", "hi"] => 'boo'
console.log(getLongestWord(["boo", "a", "I", "hi"]));

// ["one", "two", "superduper", "dog"] => 'superduper'
console.log(getLongestWord(["one", "two", "superduper", "dog"]));

// ["hello", "world"] => 'hello'
console.log(getLongestWord(["hello", "world"]));